keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1210,230 +1099,251 @@ server-side-apply (SSA) simpler. Patching individual list elements is not
 needed and there is a single owner.
 
 ```go
-// ResourceSlice provides information about available
-// resources on individual nodes.
+// One or more slices represent a pool of devices managed by a given driver.
+// How many slices the driver uses to publish that pool is driver-specific.
+// Each device in a given pool must have a unique name.
+//
+// The slice in which a device gets published may change over time. The unique identifier
+// for a device is the tuple `<driver name>/<pool name>/<device name>`. Driver name
+// and device name don't contain slashes, so it is okay to concatenate them
+// like this in a string with a slash as separator. The pool name itself may contain
+// additional slashes.
+//
+// Whenever a driver needs to update a pool, it bumps the pool generation number
+// and updates all slices with that new number and any new device definitions. A consumer
+// must only use device definitions from slices with the highest generation number
+// and ignore all others.
+//
+// If necessary, a consumer can check the number of total devices in a pool (included
+// in each slice) to determine whether its view of a pool is complete.
+//
+// For devices that are not local to a node, the node name is not set. Instead,
+// the driver may use a node selector to specify where the devices are available.
 type ResourceSlice struct {
     metav1.TypeMeta
     // Standard object metadata
+    // +optional
     metav1.ObjectMeta
 
-    // NodeName identifies the node which provides the resources
-    // if they are local to a node.
-    //
-    // A field selector can be used to list only ResourceSlice
-    // objects with a certain node name.
-    NodeName string
+    Spec ResourceSliceSpec
+
+    Status ResourceSliceStatus
+}
 
+type ResourceSliceSpec struct {
     // DriverName identifies the DRA driver providing the capacity information.
     // A field selector can be used to list only ResourceSlice
     // objects with a certain driver name.
+    //
+    // Must be a DNS subdomain and should end with a DNS domain owned by the
+    // vendor of the driver.
```

Comment:
I'm still unsure what `+required` is good for besides clarity because I've not seen an effect on generated code, but I agree that it makes sense to add it for the sake of clarity.

I'm adding a TODO for the next update and the implementation.